routers/user/home.go

```diff
@@ -350,6 +350,22 @@ func ShowSSHKeys(ctx *context.Context, uid int64) {
 	ctx.PlainText(200, buf.Bytes())
 }
 
+// ShowGPGKeys output all the public GPG keys of user by uid
+func ShowGPGKeys(ctx *context.Context, uid int64) {
+	keys, err := models.ListGPGKeys(uid)
+	if err != nil {
+		ctx.ServerError("ListGPGKeys", err)
+		return
+	}
+
+	var buf bytes.Buffer
+	for i := range keys {
+		buf.WriteString(keys[i])
+		buf.WriteString("\n")
```

Comment:
If you check GitHub's behaviour with multiple keys, it's not just concatenating the `BEGIN PGP PUBLIC KEY BLOCK` blocks, but instead merging multiple keys into a single base64 block. Not sure how it's done, but if we want to keep compatibilty, we should also do that.

Reply:
Actually it seems rather simple to merge with the gpg tool. Add both keys and then do a `gpg --armor --export` without any additional arguments. This will output a merged block containing all the keys. Hopefully this is possible in pure go too.

Reply:
I haven't tested but it migth https://sourcegraph.com/github.com/keybase/go-crypto@670ebd3adf7a737d69ffe83a777a8e34eadc1b32/-/blob/openpgp/ecdh_test.go#L343